pkg/analyze/data_source.go

```diff
@@ -0,0 +1,78 @@
+package analyze
+
+import (
+	scyllav1listers "github.com/scylladb/scylla-operator/pkg/client/scylla/listers/scylla/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"reflect"
+)
+
+type DataSource interface {
+	ServiceLister() corev1listers.ServiceLister
+	SecretLister() corev1listers.SecretLister
+	ConfigMapLister() corev1listers.ConfigMapLister
+	ServiceAccountLister() corev1listers.ServiceAccountLister
+	ScyllaClusterLister() scyllav1listers.ScyllaClusterLister
+
+	AddShutdownBehaviour(shutdownBehaviour func())
+	Shutdown()
+}
+
+type dataSource struct {
+	listers            map[reflect.Type]interface{}
+	shutdownBehaviours []func()
+}
+
+func NewDataSource(
+	serviceLister corev1listers.ServiceLister,
+	secretLister corev1listers.SecretLister,
+	configMapLister corev1listers.ConfigMapLister,
+	accountLister corev1listers.ServiceAccountLister,
+	clusterLister scyllav1listers.ScyllaClusterLister,
+) DataSource {
+
+	listersMap := make(map[reflect.Type]interface{})
+	listersMap[reflect.TypeOf(secretLister)] = secretLister
+	listersMap[reflect.TypeOf(configMapLister)] = configMapLister
+	listersMap[reflect.TypeOf(accountLister)] = accountLister
+	listersMap[reflect.TypeOf(clusterLister)] = clusterLister
+	listersMap[reflect.TypeOf(serviceLister)] = serviceLister
+
+	return &dataSource{
+		listers:            listersMap,
+		shutdownBehaviours: make([]func(), 0),
+	}
+}
```

Comment:
I don't understand the idea behind this - if you already have the listers, why do you need this constructor at all? It effectively becomes a wrapper?

Reply:
done